presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -127,6 +130,52 @@ private static SubPlan analyzeGroupedExecution(Session session, Metadata metadat
         return new SubPlan(fragment, result.build());
     }
 
+    private SubPlan reassignPartitioningHandleIfNecessary(Session session, Metadata metadata, SubPlan subPlan)
+    {
+        return reassignPartitioningHandleIfNecessaryHelper(session, metadata, subPlan, subPlan.getFragment().getPartitioning());
+    }
+
+    private SubPlan reassignPartitioningHandleIfNecessaryHelper(Session session, Metadata metadata, SubPlan subPlan, PartitioningHandle newOutputPartitioningHandle)
+    {
+        PlanFragment fragment = subPlan.getFragment();
+
+        PartitioningHandleReassigner partitioningHandleReassigner = new PartitioningHandleReassigner(fragment.getPartitioning(), metadata, session);
+        PlanNode newRoot;
```

Comment:
initialize it to `fragment.getRoot()`, invert the condition in below if and remove else. Then it will follow the pattern on 152.